smart-contracts/contracts/RelayPool.sol

```diff
@@ -302,9 +346,11 @@ contract RelayPool is ERC4626, Ownable {
   // Note: a previous version used the full balance of assets.
   //       This creates a vulnerability where a 3rd party can inflate
   //       the share price and use that to capture the value created.
-  function depositAssetsInYieldPool(uint256 amount) internal {
+  function depositAssetsInYieldPool(
+    uint256 amount
+  ) internal returns (uint256 receivedShares) {
     ERC20(asset).approve(yieldPool, amount);
-    ERC4626(yieldPool).deposit(amount, address(this));
+    receivedShares = ERC4626(yieldPool).deposit(amount, address(this));
     emit AssetsDepositedIntoYieldPool(amount, yieldPool);
```

Comment:
You are changing the signature of this function here. Not sure that is a good idea at this stage!

Reply:
But maybe that is fine